src/azure/azureController.ts

```diff
@@ -239,6 +249,35 @@ export class AzureController {
 		}
 	}
 
+	/**
+	 * Returns Azure subscriptions with tenant and token for each given account
+	 */
+	public async getAccountSessions(account: IAccount): Promise<mssql.IAzureAccountSession[]> {
+		try {
+			let subscriptions: mssql.IAzureAccountSession[] = [];
+			const tenants = <Tenant[]>account.properties.tenants;
+			for (const tenantId of tenants.map(t => t.id)) {
+				const token = await this.getAccountSecurityToken(account, tenantId, providerSettings.resources.azureManagementResource);
+				const subClient = this.getSubscriptionClient(token);
+				const newSubPages = await subClient.subscriptions.list();
+				const array = await azureUtils.getAllValues<Subscription, mssql.IAzureAccountSession>(newSubPages, (nextSub) => {
+					return {
+						subscription: nextSub,
+						tenantId: tenantId,
+						account: account,
+						token: token
+					};
+				});
+				subscriptions = subscriptions.concat(array);
+			}
+
+			return subscriptions.sort((a, b) => (a.subscription.displayName || '').localeCompare(b.subscription.displayName || ''));
+		} catch (error) {
+
+			return [];
```

Comment:
Why are you returning an empty array here? Seems reasonable to let the exception bubble up so the caller knows something went wrong.